xonsh/completers/commands.py

```diff
@@ -105,27 +108,87 @@ def complete_end_proc_keywords(command_context: CommandContext):
 class ModuleMatcher:
     """Reusable module matcher. Can be used by other completers like Python to find matching script completions"""
 
-    def __init__(self, base: str):
+    extensions = (".py", ".xsh")
+
+    def __init__(self, base: str, extra_paths: "list[str]"):
+        """Helper class to search and load Python modules
+
+        Parameters
+        ----------
+        base
+            namespace package
+        extra_paths
+            extra search paths to use if finding module on namespace package fails
+        """
         # list of pre-defined patterns. More can be added using the public method ``.wrap``
         self._patterns: tp.Dict[str, str] = {
             r"\bx?pip(?:\d|\.)*(exe)?$": "pip",
         }
         self._compiled: tp.Dict[str, tp.Pattern] = {}
         self.contextual = True
         self.base = base
+        # unique but maintain order
+        self._paths = OrderedDict([(pth, None) for pth in extra_paths])
```

Comment:
This can be a regular `dict`, I think (and still be ordered)

Reply:
Yes I considered but it doesn't feel right putting it though. It will be explicit that we follow the path order for preferring